### ISSUE TITLE
Save model's weights for each epoch

### ISSUE THREAD

**Issue by laetitialalla** [azavea/raster-vision#1720]
## 🚀 Feature

Currently, during training, the weights are saved after every epoch to `last-model.pth` in the folder `training` (i.e. the file is overwritten after every epoch). Once the training is complete, a bundle is created in the folder `bundle` and `last-model.pth` file is copied to the bundle as `model.pth`.

However, it would be interesting to keep the weights of the model for **all the epochs**.  
If we define the "best" model as the model for which a chosen metric was the highest (like map50 for object detection, on the validation set), this is not necessarily the "last" model.  

## Motivation
This is related to the discussion #1710 (model fine-tuning for an object detection task)  

## Pitch
2 propositions : 
* keep the weights at each epoch, in order to select the best model at the end of the training. 
* keep the weights of the best model only (based on some metric), even if the file is overwritten after every epoch (but that would mean add the relevant metric as a parameter)

## Alternatives

So far, the only alternative is to manually kill the job in the middle of training, if we monitor a decrease in the metric. In that case, the weights from the last completed epoch will be in `last-model.pth`.

## Additional context
This being said, as [AdeelH](https://github.com/AdeelH) wrote in #1710, one should be careful about how much stock to put in the model's performance on the validation set. If your validation set is too small or not representative of the test set or the data in production, then you run the risk of overfitting the validation set.
